pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentOnlineOfflineStateModelFactory.java

```diff
@@ -200,22 +196,11 @@ public void onBecomeDroppedFromOffline(Message message, NotificationContext cont
       _logger.info("SegmentOnlineOfflineStateModel.onBecomeDroppedFromOffline() : " + message);
       String tableNameWithType = message.getResourceName();
       String segmentName = message.getPartitionName();
-
-      // This method might modify the file on disk. Use segment lock to prevent race condition
-      Lock segmentLock = SegmentLocks.getSegmentLock(tableNameWithType, segmentName);
```

Comment:
We still need to lock the segment because it will modify the file

Reply:
locked inside dropSegment() method, like the other methods reloadSegment or so.